File: app/src/main/java/com/skydoves/medaldemo/ExampleActivity.java
/*
 * Copyright (C) 2017 skydoves
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.skydoves.medaldemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import com.skydoves.medal.MedalAnimation;

public class ExampleActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MedalAnimation medalAnimation_tv = new MedalAnimation.Builder()
                .setSpeed(3000)
                .setTurn(1)
                .build();

        TextView textView = (TextView)findViewById(R.id.textView_title);
        textView.startAnimation(medalAnimation_tv);

        MedalAnimation medalAnimation = new MedalAnimation.Builder()
                .setDirection(MedalAnimation.LEFT)
                .setSpeed(4200)
                .setTurn(4)
                .build();

        ImageView imageView = (ImageView)findViewById(R.id.badge);
        imageView.startAnimation(medalAnimation);
    }
}

File: app/src/main/java/com/skydoves/medaldemo/ExampleActivity.kt
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2017 skydoves
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.skydoves.medaldemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.skydoves.medal.MedalAnimation
import com.skydoves.medal.MedalDirection
import com.skydoves.medal.medalAnimation
import kotlinx.android.synthetic.main.activity_main.*

class ExampleActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val medalAnimationTextView = MedalAnimation.Builder().setSpeed(3000).setTurn(1).build()
    medalAnimationTextView.startAnimation(textView_title)

    val medalAnimation = medalAnimation {
      direction = MedalDirection.LEFT
      speed = 4200
      turn = 4
    }

    badge.startAnimation(medalAnimation)
  }
}
